src/main/java/com/aws/greengrass/util/ProxyUtils.java

```diff
@@ -228,6 +228,12 @@ public static void setDeviceConfiguration(DeviceConfiguration deviceConfiguratio
      *
      * @return httpClient built with a ProxyConfiguration, if a proxy is configured, otherwise
      *         a default httpClient
+     *
+     * @deprecated This method creates potential memory leak situations when wrapped by an SDK client.
```

Comment:
Instead of saying 'potential memory leak situation' we could call out the exact behavior that leads to the leak

Reply:
updated